pkg/spdx/parser.go

```diff
@@ -81,46 +84,73 @@ func OpenDoc(path string) (doc *Document, err error) {
 // parseJSON parses an SPDX document encoded in json
 // nolint:gocyclo
 func parseJSON(file *os.File) (doc *Document, err error) {
-	jsonDoc := &spdxJSON.Document{
-		CreationInfo: spdxJSON.CreationInfo{
-			Creators: []string{},
-		},
-		DocumentDescribes:    []string{},
-		Files:                []spdxJSON.File{},
-		Packages:             []spdxJSON.Package{},
-		Relationships:        []spdxJSON.Relationship{},
-		ExternalDocumentRefs: []spdxJSON.ExternalDocumentRef{},
-	}
+	var jsonDoc interface{}
 
 	// Read the SPDX doc into the json struct
 	var data []byte
 	data, err = os.ReadFile(file.Name())
 	if err != nil {
 		return nil, fmt.Errorf("reading SBOM file: %w", err)
 	}
-	if err := json.Unmarshal(data, jsonDoc); err != nil {
-		return nil, fmt.Errorf("parsing SBOM json: %w", err)
+
+	var spdxVersion string
+
+	if bytes.Contains(data, []byte("SPDX-2.3")) {
+		doc := spdx23JSON.Document{
+			CreationInfo: spdx23JSON.CreationInfo{
+				Creators: []string{},
+			},
+			DocumentDescribes:    []string{},
+			Files:                []spdx23JSON.File{},
+			Packages:             []spdx23JSON.Package{},
+			Relationships:        []spdx23JSON.Relationship{},
+			ExternalDocumentRefs: []spdx23JSON.ExternalDocumentRef{},
+		}
+		if err := json.Unmarshal(data, &doc); err != nil {
+			return nil, fmt.Errorf("parsing SBOM json: %w", err)
+		}
+		spdxVersion = "2.3"
+		jsonDoc = doc
+	} else {
+		doc := spdx22JSON.Document{
+			CreationInfo: spdx22JSON.CreationInfo{
+				Creators: []string{},
+			},
+			DocumentDescribes:    []string{},
+			Files:                []spdx22JSON.File{},
+			Packages:             []spdx22JSON.Package{},
+			Relationships:        []spdx22JSON.Relationship{},
+			ExternalDocumentRefs: []spdx22JSON.ExternalDocumentRef{},
+		}
+		if err := json.Unmarshal(data, &doc); err != nil {
+			return nil, fmt.Errorf("parsing SBOM json: %w", err)
+		}
+		spdxVersion = "2.2"
+		jsonDoc = doc
 	}
 
+	jsonValue := reflect.ValueOf(jsonDoc)
```

Comment:
Maybe having a real interface for the types would be great here. This helps avoid using reflection. I can take care of it as a follow-up, I'm not blocking on it.